### DISCUSSION TITLE
docker buildx does not uninstall and keep been used as default

### DISCUSSION THREAD

**Discussion by GabrielaRsanches** [docker/buildx#2142] [Q&A] [answered]
I have created a new container using buildx to use multiplatforms. Now my app is using this buildx container as default and it does not work with private repositories. This is causing my app not to build. I tried:
- rm the containes
- deleted all images related to it
- deleted all volumes related to it
- export DOCKER_BUILDKIT=0    (to not use buildx as default)
- echo 'export DOCKER_BUILDKIT=0' >> ~/.bashrc                                                                                                       
source ~/.bashrc  (to persist to all terminals)
- docker rm -f container-built-with-buildx
- docker buildx uninstall
try to rebuild and still getting this container to be created by buildx !

Don't know what else to do to make my program to NOT use buildx to stop getting this container to be created automatically when building the app

**Comment by crazy-max** [comment]
> I have created a new container using buildx

What command did you type to create the builder? I assume you used the command `docker buildx create`?

> keep been used as default

I guess you also set `--use` flag with `docker buildx create` command? If that's the case it marks this builder as default.

> docker buildx uninstall

This will just remove the alias of `docker build` > `docker buildx build`, not the builder, if that was your intention.

You can see the list of builders with `docker buildx ls`. Here you will see a `*` next to the current builder: https://docs.docker.com/engine/reference/commandline/buildx_ls/

If you want to remove a builder you have to use the `docker buildx rm` command.

**Reply by GabrielaRsanches** [reply]
Yes I used "docker buildx create" and set --use to it . 
The problem is that the two contexts I am getting are based on this setting/container I created.

```
docker buildx ls
NAME/NODE          DRIVER/ENDPOINT  STATUS   BUILDKIT             PLATFORMS
container          docker-container                               
  container0       desktop-linux    inactive                      
nice_noether *     docker-container                               
  nice_noether0    desktop-linux    running  v0.12.3              linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64, linux/arm/v7, linux/arm/v6
objective_hertz    docker-container                               
  objective_hertz0 desktop-linux    inactive                      
default            docker                                         
  default          default          running  v0.11.6+616c3f613b54 linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64, linux/arm/v7, linux/arm/v6
desktop-linux      docker                                         
  desktop-linux    desktop-linux    running  v0.11.6+616c3f613b54 linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/386, linux/mips64, linux/arm/v7, linux/arm/v6
```

This nice_noether0 container is the one I want to stop buildx from creating when trying to build my app. I removed it with docker rm but when I try do rebuild the app it is still been created

**Reply by GabrielaRsanches** [reply]
I also tried to uninstall buildx, did not work

**Reply by crazy-max** [reply, marked as answer]
Then you can either switch to `default` using `docker buildx use default` so it will build as before using the docker engine (nominal case)

Or you can remove `nice_noether` builder using `docker buildx rm nice_noether` and it should automatically switch to `default`.

**Reply by GabrielaRsanches** [reply]
IT WORKED NOW ! THANK YOU VERY MUCH

**Reply by ilyaDegtyarenko** [reply]
I tried, but an error occured: ERROR: failed to remove one or more builders

Even "-f" didn't help
